app/models/patient.rb

```diff
@@ -110,6 +108,10 @@ def vaccinated?
     last_appointment&.second_dose && last_appointment&.check_out.present?
   end
 
+  def allowed?
+    has_future_appointments? || can_schedule?
```

Comment:
If we invert this order and `can_schedule?` returns false we can avoid a db query right?

Also, nice refactoring :taco:

Reply:
Good idea, done :)